### PR TITLE
MAX! Thermostat configuration maintenance

### PR DESCRIPTION
**Pull request by marcelrv**
Implements maintenance via the configuration screen #201

Signed-off-by: Marcel Verpaalen <marcel@verpaalen.com>

### CODE REVIEW

File: addons/binding/org.openhab.binding.max/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -42,15 +42,61 @@
...
+				<label>Eco Temperature</label>
+				<description>Set the Eco Temperature.</description>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="offsetTemp" type="decimal" min="-3.5" max="3.5" step="0.5" required="false" groupName="device">
+				<label>OffSet Temperature</label>
+				<description>Set the Thermostat offset.</description>
+				<advanced>true</advanced>
 			</parameter>
+            <parameter name="maxTempSetpoint" type="decimal" min="4.5" max="30.5" step="0.5" required="false" groupName="device">
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.max/ESH-INF/thing/thing-types.xml]: you have some space/tab mixture in here

File: addons/binding/org.openhab.binding.max/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -42,15 +42,61 @@
...
+				<label>Room Id</label>
+				<description>The room name Id.</description>
+				<advanced>true</advanced>
+			</parameter>
+            -->
 			<parameter name="room" type="text" required="false" groupName="device">
 				<label>Room</label>
 				<description>The room name.</description>
+				<advanced>true</advanced>
+
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.max/ESH-INF/thing/thing-types.xml]: remove empty line

File: addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java [hunk 3]
```diff
@@ -258,6 +340,14 @@ private synchronized MaxCubeBridgeHandler getMaxCubeBridgeHandler() {
         return this.bridgeHandler;
     }
 
+    private Device getDevice() throws NullPointerException {
+        Device device = getMaxCubeBridgeHandler().getDevice(maxDeviceSerial);
+        if (device == null) {
+            throw new NullPointerException();
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: this is a rather unusual construct. Why not return null and handle that in the caller?

Comment (marcelrv) [inline addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: Yes, will check.. was some time ago I wrote that bit. Think it had to do with the caller getting bit complex/unreadable due to many if statements

**Comment by marcelrv** [comment]
@kaikreuzer thanks for the feedback. Improved it with additional commmit

**Comment by kaikreuzer** [comment]
Thanks!

**Pull request merged**